retry.go

```diff
@@ -158,6 +158,22 @@ func (r *realisClient) thriftCallWithRetries(thriftCall auroraThriftCall) (*auro
 			// Print out the error to the user
 			r.logger.Printf("Client Error: %v\n", clientErr)
 
+			// Determine if error is a temporary URL error by going up the stack
+			e, ok := clientErr.(thrift.TTransportException)
+			if ok {
+				r.logger.DebugPrint("Encountered a transport exception")
+
+				e, ok := e.Err().(*url.Error)
```

Comment:
It might be good add a link to https://golang.org/pkg/net/url/#Error , the error type already have a method `Temporary` , the logic is same as our `IsTemporary` util function, but for error from standard library, personally I always prefer to use what standard library has offered just in case some internal implementation changed (it rarely happen though)

Also since we are wrapping the error, do we use `errors.Cause` to unwrap in util function like `IsTemporary`

And for this retry behavior I think we should document it in package level or somewhere in READEME because the retry logic applies to all the methods (read&write), so people read the doc in full will notice it